src/cli/filter_arg.zig

```diff
@@ -0,0 +1,195 @@
+            } else break,
+            else => break,
+        };
+        const name_prop = json.asProperty("name") orelse continue;
+        const name = switch (name_prop.expr.data) {
+            .e_string => |n| n.data,
+            else => break,
+        };
+
+        _ = Package.processWorkspaceNamesArray(
+            workspace_map,
+            allocator,
+            log,
+            json_array,
+            &json_source,
+            prop.loc,
+            null,
+        ) catch |err| {
+            return err;
+        };
+
+        // add the root package to the workspace map, too
+        const entry = Package.WorkspaceMap.Entry{ .name = try allocator.dupe(u8, name), .version = null, .name_loc = bun.logger.Loc.Empty };
+        try workspace_map.insert(try allocator.dupe(u8, parent_trimmed), entry);
+
+        return;
+    }
+
+    // if we were not able to find a workspace root, try globbing for package.json files
+
+    var walker = Glob.BunGlobWalker{};
+    var arena = std.heap.ArenaAllocator.init(allocator);
+    const walker_init_res = try walker.init(&arena, "**/package.json", true, true, false, true, true);
+    switch (walker_init_res) {
+        .err => |err| {
+            Output.prettyErrorln("Error: {}", .{err});
+            return;
+        },
+        else => {},
+    }
+    defer walker.deinit(true);
+
+    var iter = Glob.BunGlobWalker.Iterator{ .walker = &walker };
+    const iter_init_res = try iter.init();
+    switch (iter_init_res) {
+        .err => |err| {
+            Output.prettyErrorln("Error: {}", .{err});
+            return;
+        },
+        else => {},
+    }
+    defer iter.deinit();
+
+    while (true) {
+        const next = try iter.next();
+        const path = switch (next) {
+            .err => |err| {
+                Output.prettyErrorln("Error: {}", .{err});
+                continue;
+            },
+            .result => |path| path orelse break,
+        };
+
+        const json_file = std.fs.cwd().openFile(
+            path,
+            .{ .mode = .read_only },
+        ) catch {
+            continue;
+        };
+        defer json_file.close();
+
+        const json_stat_size = try json_file.getEndPos();
+        const json_buf = try allocator.alloc(u8, json_stat_size + 64);
+        defer allocator.free(json_buf);
+
+        const json_len = try json_file.preadAll(json_buf, 0);
+        const json_source = bun.logger.Source.initPathString(path, json_buf[0..json_len]);
+
+        var parser = try json_parser.PackageJSONVersionChecker.init(allocator, &json_source, log);
+        _ = try parser.parseExpr();
+        if (!parser.has_found_name) {
+            continue;
+        }
+        const entry = Package.WorkspaceMap.Entry{ .name = try allocator.dupe(u8, parser.found_name), .version = null, .name_loc = bun.logger.Loc.Empty };
+        const dirpath = std.fs.path.dirname(path) orelse continue;
+        try workspace_map.insert(try allocator.dupe(u8, dirpath), entry);
```

Comment:
package jsons found here may or may not actually be workspace packages. If the script isn't found in the `package.json` we shouldn't print an error
<img width="447" alt="Screenshot 2024-01-17 at 4 06 37 PM" src="https://github.com/oven-sh/bun/assets/35280289/cd3be29a-5843-487a-8c39-b61e04164301">

Reply:
The `--if-present` flag accomplishes this, but maybe this should be the default, just like for pnpm.